Add Flow Tap to zsa/qmk_firmware.

getreuer opened the pull request:
Adds the new [Flow Tap](https://github.com/qmk/qmk_firmware/pull/25125) tap-hold option to zsa/qmk_firmware.

## Description

This is a cherry pick of https://github.com/qmk/qmk_firmware/pull/25125, adding Flow Tap in the ZSA fork. Documentation is excluded, since zsa/qmk_firmware doesn't have the `docs/` directory, otherwise it is exactly as in the current state of the mainline QMK develop branch.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

getreuer committed:
Add Flow Tap to zsa/qmk_firmware.

getreuer commented:
Moving this PR to draft for now. [A bug in Flow Tap](https://github.com/qmk/qmk_firmware/pull/25125#issuecomment-2816427787) has been found. I'll update here once that is fixed.

fdidron commented:
@getreuer Great, I already have a Oryx branch to support this, testing the feature right now and it works nicely. I will wait on your update for that bugfix.

getreuer commented:
Thank you, @fdidron! I have a Flow Tap revision in https://github.com/qmk/qmk_firmware/pull/25175 that's been verified to fix the reported bugs. I hope this is merged soon, I'll then incorporate it here.

getreuer committed:
Flow Tap bug fixes and enhancements.

Adds Tap Flow bug fix https://github.com/qmk/qmk_firmware/pull/25175 and
enhancement https://github.com/qmk/qmk_firmware/pull/25200.

getreuer commented:
@fdidron we've made a bug fix https://github.com/qmk/qmk_firmware/pull/25175 and enhancement https://github.com/qmk/qmk_firmware/pull/25200 to Flow Tap. This is working well and is ready to go. I've updated this PR to include these revisions.

If any adjustment is needed here to merge, let me know. Thank you!

fdidron commented:
@getreuer Thanks, we are going to start testing the Oryx integration internally and should release quite soon. This is a neat feature I'm already using daily :)

getreuer commented:
@fdidron that's exciting news! Thanks so much!
